client/src/utils/fight/arrive.ts

```diff
@@ -51,40 +54,42 @@ const arrive = async (
     void sound.play('sfx', { sprite: 'arrive' });
   }, 250 / speed.current);
 
+  // Set airborn phase
+  fighter.animation.setAirborn(true);
+
   // Move fighter to the position
   await Tweener.add({
     target: fighter.animation.container,
     duration: 0.5 / speed.current,
-    ease: Easing.linear
+    ease: Easing.linear,
+    onUpdate: (progress: number) => {
+      // Distance to the ground from 50 to 0
+      fighter.animation.container.zIndex = fighter.animation.container.y + 50 * (1 - progress);
+      // Display shadow at the right place
+      fighter.animation.updateShadow();
+    },
```

Comment:
tick calls can be quite hard on low end mobiles, it would be better to update the zIndex directly in the tweener like: `{ x, y, zIndex: ... }`

Shadows should also be handled by a single tween instead of being refresh on every tick, in the FIghterHolder class

Reply:
I will try to find solutions for this today...
But it will be a looot of tweeners ...